ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/tools/osb/mbean/ObjStoreBrowser.java

```diff
@@ -485,12 +490,12 @@ public void setExposeAllRecordsAsMBeans(boolean exposeAllLogs) {
     public synchronized void probe() throws MBeanException {
         Map<String, Collection<Uid>> currUidsForType = new HashMap<>();
 
-        // recreate every MBean so clear the cache first
-        unregisterMBeans();
-
         for (String type : getTypes())
             currUidsForType.put(type, getUids(type));
 
+        // if there are any beans in registeredMBeans that don't appear in new list and unregister them
+        unregisterRemovedUids(currUidsForType); //unregisterMBeans();
```

Comment:
I don't understand where this change comes from (which commit is it from)?

Reply:
Ah sorry, it's part of the fix for JBTM-3859 MBean for JTA transaction heuristics is imprecise

Reply:
Sorry, for some reason the commit message of one of the commits was not good and that wouldn't have helped. Thanks for finding it